app/src/main/java/ml/docilealligator/infinityforreddit/comment/ParseComment.java

```diff
@@ -165,7 +226,7 @@ private static void expandChildren(ArrayList<Comment> comments, ArrayList<Commen
             } else {
                 c.setExpanded(true);
             }
-            if (c.hasMoreChildrenFullnames() && c.getMoreChildrenFullnames().size() > c.getMoreChildrenStartingIndex()) {
+            if (c.hasMoreChildrenIds() && !c.getMoreChildrenIds().isEmpty()) {
                 //Add a load more placeholder
                 Comment placeholder = new Comment(c.getFullName(), c.getDepth() + 1, Comment.PLACEHOLDER_LOAD_MORE_COMMENTS);
```

Comment:
I'm not sure why it was done like this, but this looks overcomplicated - it should be easier to create placeholder during parsing itself and also store moreChildren in it instead of parent.

Reply:
I think it's because sometimes the load more button could not load all the `more children`.

Reply:
Thanks, now I see how it would help if comment was not just parsed. But since now the comments always come from the api, adding the placeholder can be moved to `parseCommentRecursion` and `parseMoreComment`

Reply:
[Here's](https://github.com/RSBat/Infinity-For-Reddit/compare/800-morechildren...RSBat:Infinity-For-Reddit:morechildren-placeholder?expand=1) what I'm suggesting

Reply:
> [Here's](https://github.com/RSBat/Infinity-For-Reddit/compare/800-morechildren...RSBat:Infinity-For-Reddit:morechildren-placeholder?expand=1) what I'm suggesting

Wow, by pure luck that code doesn't add placeholder directly to `newComments`. Adding it would cause duplicated "Load more comments" button because if `moreChildrenIds` is not empty, that button is not removed. And if placeholder is also added to `newComments`,  then it would be displayed and stored twice. :exploding_head: